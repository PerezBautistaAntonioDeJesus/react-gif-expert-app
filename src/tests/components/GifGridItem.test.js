import React from "react";
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en <GifGridItem />', () => {

    const img = {
        url: 'http://localhost:9090/something,jpg',
        title: 'My title',
    }
    const wrapper = shallow( <GifGridItem {...img}/> );

    test('should show the component correctly', () => { 
        
        expect( wrapper ).toMatchSnapshot();

     });



    test('should has a paragraph with the title', () => { 
        
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( img.title )
    
    });


    test('the img tag should have url in src attribute', () => { 

        const imgtag = wrapper.find('img')
        const properties = imgtag.props();

        expect( properties.src ).toBe(img.url);
        expect( properties.alt ).toBe(img.title);
    });


    test('should has the class animate__fadeIn', () => { 
        const div = wrapper.find( 'div');
        const classes = div.prop('className');
        expect( classes ).toContain( 'animate__fadeIn' );
     })
})