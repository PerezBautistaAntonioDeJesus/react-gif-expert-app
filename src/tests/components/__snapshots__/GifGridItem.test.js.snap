// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="My title"
    src="http://localhost:9090/something,jpg"
  />
  <p>
    My title
  </p>
</div>
`;
